js/deer-render.js

```diff
@@ -204,6 +204,255 @@ DEER.TEMPLATES.shadow = (obj, options = {}) => {
+                if(coll.itemListElement.filter(record => record["@id"] === obj['@id']).length){
+                    elem.innerHTML = `✔ Submitted For Review`
+                    elem.classList.remove("bg-error")
+                    elem.classList.add("bg-success")
+                }
+                else{
+                    let drawer = `
+                        <div class="statusDrawer is-hidden"> 
+                            <div style="margin-top: 1em;"> 
+                                <p> This will alert reviewers to look at this item.  You may add a comment below. </p>
+                                <textarea class="statusCommentText" style="margin-top: 1em;"></textarea>
+                                <input class="notifyReviewerBtn success" type="button" value="Notify Reviewers" style="margin-top: 1em;" />
+                            </div>
+                        </div>
+                    `
+                    elem.innerHTML = `Submit For Review ❕ ${drawer}`
+
+                    elem.addEventListener("click", e => {
+                        if(e.target.id !== "managedStatus"){
+                            //Do not close when users click on internal items
+                            return
+                        }
+                        if(elem.querySelector(".statusDrawer").classList.contains("is-hidden")){
+                            elem.querySelector(".statusDrawer").classList.remove("is-hidden")
+                        }
+                        else{
+                            elem.querySelector(".statusDrawer").classList.add("is-hidden")
+                        }
+                    })
+
+                    elem.querySelector(".notifyReviewerBtn").addEventListener("click", e => {
+                        if(!userHasRole(["dunbar_user_curator","dunbar_user_contributor","dunbar_user_reviewer"])) { return alert(`This function is limited to contributors, reviewers, and curators.`)}
+                        let proceed = confirm("This action is connected with you username.  Click OK to proceed and add your note.")
+                        if(!proceed){return}
+                        let url = DEER.URLS.CREATE
+                        let method = "POST"
+                        const commentAnno = {
+                            "@context": "http://purl.org/dc/terms",
+                            "motivation" : "commenting",
+                            "type": "Annotation",
+                            "body":{
+                                "comment":{
+                                    "type" : "Comment",
+                                    "text" : elem.querySelector("textarea").value,
+                                    "author" : DLA_USER["http://store.rerum.io/agent"]
+                                }
+                            },
+                            "target":obj["@id"]
+                        }
+                        if(obj?.comment.source.citationSource){
```

Comment:
Is this C/P error? This looks like MODS Source comment stuff I've used in other places.

Reply:
Nope.  That's some regularization the DEER does.  I need the @id of the commenting Annotation.  Here is what obj.comment looks like right there

```JSON
{
    "source": {
        "citationSource": "http://store.rerum.io/v1/id/63fe7e861267884358929277",
        "citationNote": "Composed object from DEER",
        "comment": "Learn about the assembler for this object at https://github.com/CenterForDigitalHumanities/deer"
    },
    "value": {
        "type": "Comment",
        "text": "Hi contributor.  I found that this needs changes.  Please fix and re-submit.  Thanks.",
        "author": "http://store.rerum.io/v1/id/62572ba71d974d1311abd673"
    },
    "evidence": ""
}
```